### DISCUSSION TITLE
samtools error with doctor.py

### DISCUSSION THREAD

**Discussion by gsk7757** [biostars/biostar-handbook-issues#288] [Q&A]
Hello, I tried installing and setting up the bioinfo environment as mentioned in the handbook. However, when I run doctor.py there were couple errors: samtools and fastq-dump. I tried installing samtools by using conda install -c bioconda samtools. But I have the same issue. Please help.

**Comment by ialbert** [comment]
it would be most curious to have samtools fail.

what happens when you run

    micromamba install samtools

what message do you get?

**Comment by gsk7757** [comment]
I tried installing samtools again using 

    micromamba install samtools

```
micromamba install samtools
pkgs/main/noarch                                              No change
pkgs/r/osx-64                                                 No change
pkgs/r/noarch                                                 No change
pkgs/main/osx-64                                     5.1MB @   5.3MB/s  1.0s
bioconda/osx-64                                      4.2MB @   3.8MB/s  1.1s
bioconda/noarch                                      4.7MB @   3.3MB/s  1.2s
conda-forge/noarch                                  12.1MB @   8.1MB/s  1.3s
conda-forge/osx-64                                  26.3MB @  12.2MB/s  1.9s

Pinned packages:
  - python 3.8.*


Transaction

  Prefix: /Users/sgadila/micromamba/envs/bioinfo

  All requested packages already installed


Transaction starting

Transaction finished

To activate this environment, use:

    micromamba activate bioinfo

Or to execute a single command in this environment, use:

    micromamba run -n bioinfo mycommand
```

After running doctor.py, it still shows the same error as below

    doctor.py

```

# Doctor! Doctor! Give me the news.
# Checking symptoms ...
# bwa           ... OK
# datamash      ... OK
# fastqc -h     ... OK
# hisat2        ... OK
# seqret -h     ... OK
# featureCounts ... OK
# efetch        ... OK
# esearch       ... OK
# samtools      --> ERROR
# fastq-dump    --> ERROR
# bowtie2       ... OK
# bcftools      ... OK
# seqtk         ... OK
# seqkit        ... OK
# bio           ... OK
# fastq-dump -X 1 -Z SRR1553591 --> ERROR
#
# Found some problems
#
# Unable to run: samtools, fastq-dump, fastq-dump -X 1 -Z SRR1553591
#
# Check the Biostar Handbook for installation help.
#
```

**Reply by ialbert** [reply]
if you just run samtools what happens?

samtools not working indicates some other problem - never seen such an error before - perhaps some sort of conflict with an existing installation

**Reply by ialbert** [reply]
See also the step-by-step instructions starting with step 5 as shown here:

https://www.biostarhandbook.com/step-by-step.html

**Reply by gsk7757** [reply]
Should I uninstall and reinstall everything from Step 1?

**Comment by gsk7757** [comment]
Here is the error message that popped up

samtools
dyld: Library not loaded: @rpath/libcurl.4.dylib
  Referenced from: /Users/sgadila/micromamba/envs/bioinfo/bin/samtools
  Reason: Incompatible library version: samtools requires version 13.0.0 or later, but libcurl.4.dylib provides version 11.0.0
Abort trap: 6

**Reply by ialbert** [reply]
ok, I see, I believe the problem is that the channels were listed in the wrong order and as such installation failed.

Run the uninstall:

    curl http://data.biostarhandbook.com/uninstall.sh | bash

then the installation again:

    curl http://data.biostarhandbook.com/install.sh | bash

**Reply by gsk7757** [reply]
Thank you. I tried uninstalling and reinstalling. After running doctor.py, except for fastq-dump all other were downloaded OK. 

doctor.py
# Doctor! Doctor! Give me the news.
# Checking symptoms ...
# bwa           ... OK
# datamash      ... OK
# fastqc -h     ... OK
# hisat2        ... OK
# seqret -h     ... OK
# featureCounts ... OK
# efetch        ... OK
# esearch       ... OK
# samtools      ... OK
# fastq-dump    --> ERROR
# bowtie2       ... OK
# bcftools      ... OK
# seqtk         ... OK
# seqkit        ... OK
# bio           ... OK
# fastq-dump -X 1 -Z SRR1553591 --> ERROR
#
# Found some problems
#
# Unable to run: fastq-dump, fastq-dump -X 1 -Z SRR1553591
#
# Check the Biostar Handbook for installation help.
#
(bioinfo)

**Reply by ialbert** [reply]
make sure to restart the terminal, 

also can you run the command:

    ~/bin/sratoolkit.3.0.0-mac64/bin/fastq-dump

that is where the tools should present

**Reply by gsk7757** [reply]
After running the command:
~/bin/sratoolkit.3.0.0-mac64/bin/fastq-dump

I have the following errors:
~/bin/sratoolkit.3.0.0-mac64/bin/fastq-dump
dyld: lazy symbol binding failed: Symbol not found: ____chkstk_darwin
  Referenced from: /Users/sgadila/bin/sratoolkit.3.0.0-mac64/bin/fastq-dump (which was built for Mac OS X 10.15)
  Expected in: /usr/lib/libSystem.B.dylib

dyld: Symbol not found: ____chkstk_darwin
  Referenced from: /Users/sgadila/bin/sratoolkit.3.0.0-mac64/bin/fastq-dump (which was built for Mac OS X 10.15)
  Expected in: /usr/lib/libSystem.B.dylib

Abort trap: 6
(bioinfo)

**Reply by ialbert** [reply]
that is one weird error for sure,  never seen that one either, 

try this, manually download the latest version

    mkdir -p tmp
    cd tmp
    wget https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-mac64.tar.gz
    tar zxvf sratoolkit.current-mac64.tar.gz
    sratoolkit.3.0.7-mac64/bin/fastq-dump
  
  
  did the last command run?

**Reply by gsk7757** [reply]
Same error, when I ran the last command

sratoolkit.3.0.7-mac64/bin/fastq-dump
dyld: lazy symbol binding failed: Symbol not found: ____chkstk_darwin
  Referenced from: /Users/sgadila/tmp/sratoolkit.3.0.7-mac64/bin/fastq-dump (which was built for Mac OS X 12.6)
  Expected in: /usr/lib/libSystem.B.dylib

dyld: Symbol not found: ____chkstk_darwin
  Referenced from: /Users/sgadila/tmp/sratoolkit.3.0.7-mac64/bin/fastq-dump (which was built for Mac OS X 12.6)
  Expected in: /usr/lib/libSystem.B.dylib

Abort trap: 6
(bioinfo)

**Reply by ialbert** [reply]
how up to date is your mac, 

what version of MacOS do you have? Apple-> About this Mac

**Reply by gsk7757** [reply]
My MacOS version is 10.13.6.

**Reply by ialbert** [reply]
my guess is that might be the problem, would need to be updated to a newer version, that is a six year old operating system
